### PR TITLE
[IMP] base: improve performances of commercial fields

### PR DESCRIPTION
**Pull request by benoit-phytocontrol**
This PR improve performances by reducing the number of SQL queries when writing a res.partner's commercial field.
    
Test case: write property_account_position_id on a partner with 10 children

Before this PR: 120 SQL queries
After this PR: 12 SQL queries

I tested on all versions 14.0 to 17.0:

  | Odoo 14.0 | Odoo 15.0 | Odoo 16.0 | Odoo 17.0 | Odoo master
-- | -- | -- | -- | -- | --
queries before this commit | 252 | 263 | 142 | 120 | 155
queries after this commit | 22 | 22 | 12 | 12 | 12

The difference is even greater if specific modules have added additional commercial fields

### CODE REVIEW

**Comment by Julien00859** [comment]
@robodoo delegate=kmagusiak

File: odoo/addons/base/models/res_partner.py [hunk 1]
```diff
@@ -606,15 +606,16 @@ def _commercial_sync_from_company(self):
...
-        sync_vals = commercial_partner._update_fields_values(self._commercial_fields())
+        if fields_to_sync is None:
+            fields_to_sync = self._commercial_fields()
+        sync_vals = commercial_partner._update_fields_values(fields_to_sync)
         sync_children = self.child_ids.filtered(lambda c: not c.is_company)
         for child in sync_children:
-            child._commercial_sync_to_children()
+            child._commercial_sync_to_children(fields_to_sync)
         res = sync_children.write(sync_vals)
-        sync_children._compute_commercial_partner()
```
Comment (kmagusiak) [inline odoo/addons/base/models/res_partner.py:617]: I think this was called to compute the commercial partner to flush `is_company` and `commercial_partner_id` fields.
You might want to flush `commercial_partner_id` of self._child_ids before applying this.

Comment (benoit-phytocontrol) [reply odoo/addons/base/models/res_partner.py:617]: This line dated from 2017 and was used to fix a bug on  `commercial_partner_id` compute.

I cannot reproduce the bug described in https://github.com/odoo/odoo/commit/0d68acff8e4436e55676a5016e0496b7c3aa35aa.
It was probably fixed with the new `recursive=True` attribute.

**Review by kmagusiak — APPROVED** [review]

**Comment by kmagusiak** [comment]
@robodoo r+

**Comment by ryv-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**